src/world/Server/Packets/Handlers/VehicleHandler.cpp

```diff
@@ -16,119 +16,187 @@ This file is released under the MIT license. See README-MIT for more information
+    if (!seat->canSwitchFromSeat())
         return;
 
-    const auto sourceUnit = _player->GetMapMgr()->GetUnit(srlPacket.sourceGuid.getRawGuid());
-    if (sourceUnit == nullptr)
-        return;
-
-    if (sourceUnit->getVehicleComponent() == nullptr)
+#if VERSION_STRING < Cata
+    CmsgChangeSeatsOnControlledVehicle srlPacket;
+    if (!srlPacket.deserialise(recvPacket))
         return;
 
-    if (sourceUnit->getGuid() != _player->getCurrentVehicle()->GetOwner()->getGuid())
-        return;
+    uint64_t guid = srlPacket.sourceGuid;               // current vehicle guid
+    uint64_t accessory = srlPacket.destinationGuid;     //  accessory guid
 
-    const auto destinationUnit = _player->GetMapMgr()->GetUnit(srlPacket.destinationGuid.getRawGuid());
-    if (destinationUnit == nullptr)
-        return;
+    vehicle_base->obj_movement_info = srlPacket.movementInfo;
+    int8_t seatId = srlPacket.seat;
 
-    if (destinationUnit->getVehicleComponent() == nullptr)
+    if (vehicle_base->getGuid() != guid)
         return;
 
-    if (sourceUnit->getGuid() == destinationUnit->getGuid())
+    if (!accessory)
+    {
+        GetPlayer()->callChangeSeat(seatId, seatId > 0); // prev/next
+    }
+    else if (Unit* vehUnit = GetPlayer()->GetMapMgrUnit(accessory))
     {
-        sourceUnit->getVehicleComponent()->MovePassengerToSeat(_player, srlPacket.seat);
+        if (Vehicle* vehicle = vehUnit->getVehicleKit())
+            if (vehicle->hasEmptySeat(seatId))
+                vehUnit->handleSpellClick(GetPlayer(), seatId);
     }
     else
     {
-        if (sourceUnit->getVehicleBase()->getGuid() != destinationUnit->getVehicleBase()->getGuid())
-            return;
+        if (vehicle_base->getVehicle())
+            if (vehicle_base->getVehicle()->hasEmptySeat(seatId))
+                vehicle_base->getVehicleBase()->handleSpellClick(GetPlayer(), seatId);
+    }
 
-        _player->getCurrentVehicle()->EjectPassenger(_player);
-        destinationUnit->getVehicleComponent()->AddPassengerToSeat(_player, srlPacket.seat);
+#else
+    static MovementStatusElements const accessoryGuid[] =
+    {
+        MSEExtraInt8,
+        MSEGuidBit2,
+        MSEGuidBit4,
+        MSEGuidBit7,
+        MSEGuidBit6,
+        MSEGuidBit5,
+        MSEGuidBit0,
+        MSEGuidBit1,
+        MSEGuidBit3,
+        MSEGuidByte6,
+        MSEGuidByte1,
+        MSEGuidByte2,
+        MSEGuidByte5,
+        MSEGuidByte3,
+        MSEGuidByte0,
+        MSEGuidByte4,
+        MSEGuidByte7,
+    };
+
+    ExtraMovementStatusElement extra(accessoryGuid);
+    MovementInfo movementInfo;
+    movementInfo.readMovementInfo(recvPacket, recvPacket.GetOpcode(), &extra);
+    vehicle_base->obj_movement_info = movementInfo;
+
+    ObjectGuid accessory = extra.Data.guid;
+    int8_t seatId = extra.Data.byteData;
+
+    if (vehicle_base->getGuid() != movementInfo.guid)
+        return;
+
+    if (!accessory)
+        GetPlayer()->callChangeSeat(seatId, seatId > 0); // prev/next
```

Comment:
Use braces in 'if' statement if you used them in 'else' statement

Reply:
cant find this one